### ISSUE TITLE
Evaluate just one new image

### ISSUE THREAD

**Issue by pmfcdb** [david-gpu/srez#50]
Hi, I install and run your project ok, but know I would to restore the graph model and evaluate single images that are not in the dataset.

The restore part you already have there, but I 'm struggling with the prediction of a single image.

Can you give some pointers?

Thks,

**Comment by pmfcdb** [comment]
So here is my solution (hack), I copy srez_main.py and created a new script and then I created a new function based on _demo function, called _evaluate(). This is the function(_evaluate()) that I call in main.

def _evaluate():
    # Load checkpoint
    if not tf.gfile.IsDirectory(FLAGS.checkpoint_dir):
        raise FileNotFoundError("Could not find folder `%s'" % (FLAGS.checkpoint_dir,))

    # Setup global tensorflow state
    sess, summary_writer = setup_tensorflow()

    # Prepare directories
    Tfilenames = prepare_dirs(delete_train_dir=False)
    
    features, labels = srez_input.setup_inputs(sess, Tfilenames)
 
    # Create and initialize model
    [gene_minput, gene_moutput,
     gene_output, gene_var_list,
     disc_real_output, disc_fake_output, disc_var_list] = \
            srez_model.create_model(sess, features, labels)

    # Restore variables from checkpoint_dir
    saver = tf.train.Saver()
    filename = 'checkpoint_new.txt'
    filename = os.path.join(FLAGS.checkpoint_dir, filename)
    saver.restore(sess, filename)

    print("RESTORE model")

    #region prediction test
    predict_restore = gene_moutput

# Prepare directories
    filenames = prepare_dirs(delete_train_dir=False)

# here you can put your images, just keep the 16 size
    test_filenames  = ['default.jpg', 'testdataset/default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg','default.jpg', 'default.jpg']

    test_features,  test_labels  = srez_input.setup_inputs(sess, test_filenames)
    test_img4_input, test_img4_original  = sess.run([test_features, test_labels])

   
    test_img5 = test_img4_input.eval(session=sess)
    feed_dict={gene_minput:test_img5}
    prob =sess.run(predict_restore,feed_dict)
    #endregion prediction test

    td = TrainData(locals())
    srez_train._summarize_progress(td, test_img4_input, test_img4_original, prob, 000003, 'out')

    print("Finish EVALUATING")
